packages/trpc/server/routers/viewer/eventTypes/update.handler.ts

```diff
@@ -478,6 +488,41 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
     }
   }
 
+  // Logic for updating `descriptionTranslations`
+  // user has no description translations OR user is changing the description
+  const descriptionTranslationsNeeded = eventType.descriptionTranslations.length === 0 || newDescription;
+  const description = newDescription ?? eventType.description;
+
+  if (
+    ctx.user.organizationId &&
+    autoTranslateDescriptionEnabled &&
+    descriptionTranslationsNeeded &&
+    description
+  ) {
+    // TODO: we want to support the other locales in Locales enum + turn this into a background job using Bull Queue
+    const targetLocales = (["en", "es", "de", "pt", "fr", "it", "ar", "zh", "ru"] as const).filter(
+      (locale) => locale !== ctx.user.locale
+    );
+
+    // First, get all translations in parallel
+    const translatedTexts = await Promise.all(
+      targetLocales.map((targetLocale) =>
+        ReplexicaService.localizeText(description, ctx.user.locale, targetLocale)
+      )
+    );
+
+    // Then create all translations in a single DB call
+    await EventTypeTranslationRepository.createManyDescriptionTranslations(
```

Comment:
Because of this added logic, event type edit will take longer if user toggles this feature on. We should turn this into a background job (in a separate PR).